### PR TITLE
Extended rolling ball example with ECG data (1D) 

### LINKED ISSUE #5255: Add rolling ball example for ECG data
## Description

The demo I made for this response in the PyWavelets google group would be nice to include in our gallery to show that the "n-dimensional" implementation of rolling ball allows application to signals as well:
https://groups.google.com/g/pywavelets/c/b3ErI_GmNBo

Fortunately, the short ECG waveform used for the example there is already bundled by PyWavelets.

### PR DESCRIPTION
**Pull request by sebgoti**
## Description

closes #5255 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 1]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
 plt.show()
+
+######################################################################
+# 1D Signal Filtering
+# -----------------
+#
+# As another example of the n-dimensional feature of
+# ``rolling_ball``, we show an implementation for 1D data. Here,
+# we are interested in removing the background signal of an ECG waveform
+# in order to detect significant peaks. Smoother peaks van be removed
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:306]:
```suggestion
# to detect significant peaks. Smoother peaks can be removed
```

Comment (sciunto) [reply doc/examples/segmentation/plot_rolling_ball.py:306]: What do you mean by significant?

**Review by sciunto — CHANGES_REQUESTED** [review]
I found a typo and made suggestion. Please, could you share a screenshot?

File: doc/examples/segmentation/plot_rolling_ball.py (hunk 1 above)
Comment (sebgoti) [reply doc/examples/segmentation/plot_rolling_ball.py:306]:
```suggestion
# to detect significant peaks (higher values than the local baseline). 
# Smoother peaks can be removed
```

Comment (sebgoti) [reply doc/examples/segmentation/plot_rolling_ball.py:306]: Is the additional explanation for significant peaks (inside parenthesis) better? Thanks for the corrections.

Comment (sciunto) [reply doc/examples/segmentation/plot_rolling_ball.py:306]: It's better. Perhaps prominent peaks is rather concise and precise.

Comment (sebgoti) [reply doc/examples/segmentation/plot_rolling_ball.py:306]: screenshot
 Hi @sciunto, sorry for the inconvenience but I applied your corrections and fix a couple of errors and then pushed to the issue5255 branch of my fork but for some reason this is not being updated in the current PR. Do you have any experience with this issue?

Comment (sciunto) [reply doc/examples/segmentation/plot_rolling_ball.py:306]: That must appear automatically. Double check the branch with git branch, the remote with git remote, and the history with git log. Try to push again.

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 2]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
+# Smoother peaks can be removed with smaller values of the radius.
+
+x = pywt.data.ecg()
+background = restoration.rolling_ball(x, radius=80)
+plt.figure()
+plt.plot(x)
+plt.plot(x - background)
+background2 = restoration.rolling_ball(x, radius=10)
+plt.plot(x - background2)
+plt.legend(('original', 'radius 80', 'radius 10'))
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:316]:
```suggestion
plt.legend()
```

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 3]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
+# As another example of the n-dimensional feature of
+# ``rolling_ball``, we show an implementation for 1D data. Here,
+# we are interested in removing the background signal of an ECG waveform
+# to detect prominent peaks (higher values than the local baseline).
+# Smoother peaks can be removed with smaller values of the radius.
+
+x = pywt.data.ecg()
+background = restoration.rolling_ball(x, radius=80)
+plt.figure()
+plt.plot(x)
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:312]:
```suggestion
plt.plot(x, label='original')
```

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 4]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
+plt.plot(x - background)
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:313]:
```suggestion
plt.plot(x - background, label='radius=80')
```

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 5]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
+background2 = restoration.rolling_ball(x, radius=10)
+plt.plot(x - background2)
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:315]:
```suggestion
plt.plot(x - background2, label='radius=10')
```

File: doc/examples/segmentation/plot_rolling_ball.py [hunk 6]
```diff
@@ -294,3 +295,23 @@ def plot_result(image, background):
...
+# we are interested in removing the background signal of an ECG waveform
+# to detect prominent peaks (higher values than the local baseline).
+# Smoother peaks can be removed with smaller values of the radius.
+
+x = pywt.data.ecg()
+background = restoration.rolling_ball(x, radius=80)
+plt.figure()
+plt.plot(x)
+plt.plot(x - background)
+background2 = restoration.rolling_ball(x, radius=10)
```
Comment (sciunto) [inline doc/examples/segmentation/plot_rolling_ball.py:314]: Please, move this line right below `background = ` to gather similar actions.

**Review by sciunto — APPROVED** [review]
Few minor comments, other than that, it looks good to me.

**Review by grlee77 — APPROVED** [review]
Thanks @sebgoti, this looks good to me!

**Comment by sebgoti** [comment]
Thank you both @sciunto and @grlee77 and sorry for the delays, it was my first contribution ever so I was a bit in learning mode.

**Comment by sciunto** [comment]
many thanks to you! :)

**Pull request merged**